packages/cli/src/commands/login/future.ts

```diff
@@ -120,46 +115,59 @@ export async function login(client: Client): Promise<number> {
             );
             continue;
           default:
-            return tokenError.cause;
+            return tokensError.cause;
         }
       }
 
-      if (tokenError) return tokenError;
+      if (tokensError) return tokensError;
+
+      // If we get here, we throw away any possible token errors like polling, or timeouts
+      error = undefined;
 
       o.print(eraseLines(2));
 
       // user is not currently authenticated on this machine
       const isInitialLogin = !client.authConfig.token;
 
-      // Save the user's authentication token to the configuration file.
-      client.authConfig.token = token.access_token;
-      error = undefined;
+      const [inspectError, payload] = await inspectToken(tokens.access_token);
 
-      const [accessTokenError, accessToken] = await verifyJWT(
-        token.access_token
-      );
+      if (inspectError) return inspectError;
 
-      if (accessTokenError) {
-        return accessTokenError;
-      }
+      o.debug('access_token inspected');
+
+      client.updateAuthConfig({
+        token: tokens.access_token,
+        type: 'oauth',
+        expiresAt: Math.floor(Date.now() / 1000) + tokens.expires_in,
+      });
+
+      if (payload.team_id) o.debug('Current team updated');
+      else o.debug('Current team deleted');
+
+      client.updateConfig({ currentTeam: payload.team_id });
+
+      if (tokens.refresh_token) {
+        const [inspectError, payload] = await inspectToken(
+          tokens.refresh_token
+        );
+
+        if (inspectError) return inspectError;
 
-      o.debug('access_token verified');
+        o.debug('refresh_token inspected');
 
-      if (accessToken.team_id) {
-        o.debug('Current team updated');
-        client.config.currentTeam = accessToken.team_id;
-      } else {
-        o.debug('Current team deleted');
-        delete client.config.currentTeam;
+        client.updateAuthConfig({
+          refreshToken: tokens.refresh_token,
+          refreshTokenExpiresAt: payload.exp,
```

Comment:
IMO, we could ignore `payload.exp` and remove the refresh token inspection above:

```ts
        const [inspectError, payload] = await inspectToken(
          tokens.refresh_token
        );
```